File: exam/src/documents/Report.java
package documents;

import java.util.HashMap;

public class Report extends Document {

    public Report(String filename, HashMap<String, String> metadata) {
        super(filename, metadata);
    }

    @Override
    public boolean acceptedSize() {
        return Float.parseFloat(this.metadata.get("size")) < 4.00;
    }

    @Override
    public boolean acceptedExtensions() {
        return this.filename.toLowerCase().endsWith(".xlsx") ||  this.filename.toLowerCase().endsWith(".xls");
    }

    @Override
    public boolean hasRequiredMetadata() {
        return this.metadata.containsKey("fiscalYear") && this.metadata.containsKey("department");
    }

    @Override
    public boolean isValid() {
        return super.isValid() && acceptedSize() && acceptedExtensions() && hasRequiredMetadata();
    }
}

File: exam/src/objects/Report.ts
import { Document } from "./Document";

export class Report extends Document {
    private fiscalYear: number = 0;
    private department: string = "";

    constructor (filename: string, metadata: Record <string, string>) {
        super(filename, metadata);
        this.fiscalYear = Number.parseInt(metadata["fiscalYear"]);
        this.department = metadata["department"];
    }

    public getMetadata(): Record<string, string> {
        let metadata: Record<string,string> = {};

        metadata["fiscalYear"] = this.fiscalYear.toString();
        metadata["department"] = this.department;
        return metadata;
    }

    public getFiscalYear(): number {
        return this.fiscalYear;
    }

    public getDepartment(): string {
        return this.department;
    }

    public setFiscalYear(fiscalYear: number): void {
        this.fiscalYear = fiscalYear;
    }

    public setDepartment(department: string): void {
        this.department = department;
    }
}